plugins/inventory/nb_inventory.py

```diff
@@ -820,6 +856,32 @@ def get_region_for_site(site):
         # Dictionary of site id to region id
         self.sites_region_lookup = dict(map(get_region_for_site, sites))
 
+    # Note: depends on the result of refresh_sites_lookup for self.sites_with_prefixes
+    def refresh_prefixes(self):
+        # Pull all prefixes defined in NetBox
+        url = self.api_endpoint + "/api/ipam/prefixes"
+
+        if self.fetch_all:
+            prefixes = self.get_resource_list(url)
+        else:
+            prefixes = self.get_resource_list_chunked(
+                api_url=url,
+                query_key="site",
+                query_values=list(self.sites_with_prefixes),
+            )
+        self.prefixes_sites_lookup = defaultdict(dict)
+        self.prefixes_lookup = defaultdict(dict)
+
+        # We are only concerned with Prefixes that have actually been assigned to sites
```

Comment:
Is it possible to query the API and only get back prefixes assigned to this site, instead of looping through all the prefixes that are in the DB? My concern is that this will have bad performance on NetBox instances with lots of prefixes, similar to the bad performance I have experienced with `cables`

Reply:
It does not look like the API provides a query for "is_assigned" or similar.

My goal here was to follow the precedent that was already set by existing code. 
The optimization of grabbing only prefixes that are relevant (by being assigned to a site, is already handled by the `get_resource_list_chunked()` function.
As you said though, it is only used if the user is not using fetch_all option.
Depending on the number of sites and prefixes, assigned and unassigned in the user's database, fetch_all may or may not be faster.